pkg/codegen/pcl/utilities.go

```diff
@@ -162,3 +163,30 @@ func FixupPulumiPackageTokens(r *Resource) {
 		r.Token = "pulumi::" + name
 	}
 }
+
+// SortedFunctionParameters returns a list of properties of the input type from the schema
+// for an invoke function call which has multi argument inputs. We assume here
+// that the expression is an invoke which has it's args (2nd parameter) annotated
+// with the original schema type. The original schema type has properties sorted.
+// This is important because model.ObjectType has no guarantee of property order.
+func SortedFunctionParameters(expr *model.FunctionCallExpression) []*schema.Property {
```

Comment:
I changed the implementation to return an empty array instead of `nil` because if `MultiArgumentInputs` then the array of properties is must be non-empty which relies on invoke annotations. 

> Could we just spread multi-argument invoke parameters into the Signature.Paremeters field? Then retrieving the sorted parameters would be as simple as express.Signature.Parameters[1:].

That would change the shape of the invoke in PCL too and I am more inclined to leave as an object that corresponds with the schema inputs object.